### ISSUE TITLE
Webpack cannot resolve i18n.config and lazy translations

### ISSUE THREAD

**Issue by YarikSWT** [nuxt-modules/i18n#2115]
### Environment

- Operating System: `Darwin`
- Node Version:     `v16.16.0`
- Nuxt Version:     `3.5.1`
- Nitro Version:    `2.4.1`
- Package Manager:  `npm@8.11.0`
- Builder:          `webpack`
- User Config:      `builder`, `modules`, `i18n`
- Runtime Modules:  `@nuxtjs/i18n@8.0.0-beta.12`
- Build Modules:    `-`

### Reproduction

https://stackblitz.com/edit/nuxt-starter-bk1a9i?file=nuxt.config.ts

### Describe the bug

@nuxtjs/i18n module requires i18n.config file, when I use Vite as a builder everything works, but I need to use Webpack. When I tried to build nuxt3 project I got error caused by  the inability to resolve the path to the required config file. Same problem I got when I trying to add lazy translations. Webpack cannot resolve them.

### Logs

```shell
ERROR  in ./.nuxt/i18n.options.mjs                                                                                          

Module not found: Error: Package path ./__config__?target=../i18n.config.ts&hash=8468eb7d is not exported from package /home/projects/nuxt-starter-bk1a9i/node_modules/@nuxtjs/i18n (see exports field in /home/projects/nuxt-starter-bk1a9i/node_modules/@nuxtjs/i18n/package.json)
```

**Comment by ProEvv** [comment]
I have same problem

**Comment by dargmuesli** [comment]
@kazupon I marked this as p4 because I think webpack is meant to be supported by this module. Feel free to correct that triage :)

**Comment by paidge** [comment]
Lazy loading not working here. It works only for `defaultLocale `

**Comment by savyjs** [comment]
> Lazy loading not working here. It works only for `defaultLocale `

...and does not work in the upper layer as well.

**Comment by BobbieGoede** [comment]
@YarikSWT this issue should be resolved by #2265, you can try it out by installing the edge channel. I've changed your reproduction to demonstrate this [here](https://stackblitz.com/edit/nuxt-starter-4ruhoi?file=package.json,nuxt.config.ts). 

I did run into a separate issue caused by `vue-demi` and found a workaround by adding `build: { transpile: ['vue-demi'] }` to the nuxt config. 

I'll be closing this for now as it looks like the root issue has been fixed, please let me know if you're still experiencing this issue on the edge channel!
